processor/src/main/java/org/mapstruct/ap/internal/util/workarounds/SpecificCompilerWorkarounds.java

```diff
@@ -134,7 +140,8 @@ static TypeMirror asMemberOf(Types typeUtils, ProcessingEnvironment env, Version
             catch ( IllegalArgumentException e ) {
                 lastException = e;
                 if ( versionInformation.isEclipseJDTCompiler() ) {
-                    result = EclipseAsMemberOfWorkaround.asMemberOf( env, containing, element );
+                    result = (TypeMirror) ClassLoaderHelper.getAsMemberOf( element.getClass().getClassLoader() )
```

Comment:
should we check for the Eclipse JDT compiler version and only then perform this trick?